UnityProject/Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -1520,63 +1520,40 @@ static void ClearSpawnedDirtyBits()
         }
 
         // helper function to broadcast the world to a connection
-        static void BroadcastToConnection(NetworkConnectionToClient connection)
-        {
-            // for each entity that this connection is seeing
-            //CUSTOM UNITYSTATION CODE// for i Faster
-            var ListCount = connection.observingList.Count;
-            for (int i = 0; i < ListCount; i++)
-            {
-	            var identity = connection.observingList[i];
-            // foreach (NetworkIdentity identity in connection.observing)
-            // {
-                // make sure it's not null or destroyed.
-                // (which can happen if someone uses
-                //  GameObject.Destroy instead of
-                //  NetworkServer.Destroy)
-                //CUSTOM UNITYSTATION CODE// == null = lag
-                // if (identity != null)
-                // {
-                //CUSTOM UNITYSTATION CODE//
-	            if (identity.isDirty == false)
-	            {
-		            continue;
+        static void BroadcastToConnections()
+        {
+	        //CUSTOM UNITYSTATION CODE//
+	        //Currently all players observe EVERYTHING so we can just loop through all identities and then all connections
+	        //Old mirror code was loop through all connections and then identities which was very bad for performance
+            var connectionsCount = connectionsCopy.Count;
+            foreach (var identityPair in NetworkIdentity.spawned)
+            {
+	            var identity = identityPair.Value;
+	            if (identity.isDirty == false || identity.observers == null) continue;
 
-	            }
-	            ////////////////////////////
-	            // get serialization for this entity viewed by this connection
-	            // (if anything was serialized this time)
-	            NetworkWriter serialization = GetEntitySerializationForConnection(identity, connection);
-	            if (serialization != null)
+	            for (int j = 0; j < connectionsCount; j++)
 	            {
-		            EntityStateMessage message = new EntityStateMessage
+		            var connection = connectionsCopy[j];
+
+		            //Connection disconnected
+		            if (connection.isReady == false) continue;
+
+		            if (identity.observers.ContainsKey(connection.connectionId) == false) continue;
+
+		            ////////////////////////////
+		            // get serialization for this entity viewed by this connection
+		            // (if anything was serialized this time)
+		            NetworkWriter serialization = GetEntitySerializationForConnection(identity, connection);
```

Comment:
I'm wondering if a can cache this serialization and reuse it for all the connections, as this is what is mainly draining performance too